cmd/test/cmd.go

```diff
@@ -379,3 +363,41 @@ func shouldRunInRemote(manifestDeploy *model.DeployInfo) bool {
 
 	return false
 }
+
+func doBuild(ctx context.Context, manifest *model.Manifest, svcs []string, builder builder, ioCtrl *io.Controller) (bool, error) {
+	// make sure the images used for the tests exist. If they don't build them
+	svcsToBuild := []string{}
+	for _, name := range svcs {
+		imgName := manifest.Test[name].Image
+		getImg := func(manifest *model.Manifest) string { return imgName }
+		svc, err := builder.GetSvcToBuildFromRegex(manifest, getImg)
+		if err != nil {
+			if errors.Is(err, buildv2.ErrOktetBuildSyntaxImageIsNotInBuildSection) {
+				return false, fmt.Errorf("the image '%s' is not in the build section of the manifest", imgName)
```

Comment:
I would suggest to make this error clearer, together with the suggestion above, this error can become very clear and point to the docs.

Example:

<img width="1430" alt="Screenshot 2024-06-03 at 23 14 15" src="https://github.com/okteto/okteto/assets/2318450/a95e7a64-a9f7-41ea-a487-ed5e4b0dbebe">


```suggestion
				return false, fmt.Errorf("test '%s' needs image '%s' but it's not defined in the build section of the Okteto Manifest. See: https://www.okteto.com/docs/core/okteto-variables/#built-in-environment-variables-for-images-in-okteto-registry", name, imgName)
```